### ISSUE TITLE
MySQL 실행 오류

### ISSUE THREAD

**Issue by 201432420** [wafflestudio/rookies#215]
"mysql -uroot -p" 키워드로 MySQL 서버로 연결할 때, 다음과 같은 오류가 뜹니다.

"ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)"

<img width="795" alt="Screen Shot 2020-09-29 at 2 13 35 AM" src="https://user-images.githubusercontent.com/52521749/94464741-d3649180-01f9-11eb-8aea-69ee21f9e965.png">

이틀 전만 해도 문제 없이 켜지던 MySQL 서버였기에 문제의 원인을 파악하지 못하고 해결을 하지 못한 채 더 앞으로 나아가지 못하고 있습니다. 참고로 저는 "brew install mysql"을 통해 MySQL을 설치했습니다. 

지난 3시간 동안 구글링을 통해 stackoverflow에 제시된 방법들과 터미널 키워드들을 사용해봤지만 해결하지 못했습니다. homebrew로 MySQL을 지우고 나서 재설치도 해보고, homebrew를 통해 설치한 MySQL을 제거하고 난 후, 공식 홈페이지의 DMG 파일로 설치를 시도해보기도 했습니다만 큰 소득이 없었습니다. 참고로 제 macOS 버전은 10.13.6이라 공식홈페이지 DMG 파일은 호환가능한 5.7.23 버전으로 다운받았습니다. 

혹시 비슷한 문제를 맞닥뜨리고 해결한 분이 계신다면 도움을 구하고 싶습니다.

**Comment by eldpswp99** [comment]
#99 참고해보시겠어요?

**Comment by 201432420** [comment]
아 네 참고해보겠습니다! 감사합니다.

**Comment by davin111** [comment]
@201432420 정 안되어서 다 꼬였다 싶으면, 진짜진짜 MySQL 완전 삭제(관련 directory까지 직접 다)하는 법(예를 들면 https://kim-daeyong.github.io/2019-01-15-mysql-%EC%99%84%EC%A0%84%EC%82%AD%EC%A0%9C/ 등)을 찾아서 그대로 수행해보시는 것도 추천드립니다.

**Comment by davin111** [comment]
급한 상황이실텐데 설명충스럽지만, `mysql.sock`에 대해서는 #125 에서 설명한 적이 있습니다. 저도 이런 상황 예전에 간간히 겪었던 것 같네요. 홧팅..!

**Comment by 201432420** [comment]
네, 저 사이트의 키워드들을 실행해 다 삭제하고 다시 설치해보았습니다. 그래도 같은 오류가 떳었어요 ㅠㅠ 알려주신 이슈 읽고, mysql.sock에 대해서 더 알아보도록 하겠습니다.

**Comment by davin111** [comment]
#99 이슈에서 문제 겪으셨던 @tictactoeid 께서, 여기와 저 이슈(한 곳에만 남기고 다른 곳에는 comment link를 주면 되겠지요)에 해결하셨던 법 남겨주셔도 좋을 것 같습니다..!

**Comment by davin111** [comment]
@201432420 저기 보면 `Build: cmake` 옆에 x도 의심스러운데, 저거 관련한 문제는 아닐까요?

**Comment by 201432420** [comment]
앗 그쪽으로도 한 번 찾아보겠습니다! 감사합니다.

**Comment by tictactoeid** [comment]
해결하셨나요? #99 작성자인데 저는 아직 해결 못해서 그냥 부팅시마다
```
ln -s /var/run/mysqld/mysqld.sock /tmp/mysql.sock
```
terminal에 입력하는 식으로 하고 있습니다.

**Comment by 201432420** [comment]
지난 2시간동안 작업하다 이제 드디어 MySQL이 정상적으로 돌아가네요. 

저는 어제 "ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)”를 해결하지 못한 채 잤었는데, 오늘은 왜인지 이 오류가 뜨지 않고 "NameError: name '_mysql' is not defined”가 대신 떳습니다. MySQL 서버는 이제 정상적으로 작동하고 있었고, waffle_backend 디렉토리에서 python manage.py migrate를 시도하면서 NameError가 뜬 것입니다. 이에 대해서는 어제 노트북 전원을 끄고 키는 과정에서 제가 어제 시도했던 키워드들이 적용되어서 그렇다고 짐작만 할 뿐입니다. 

저는 "NameError: name '_mysql' is not defined”가 뜨는 이유로 MySQL를 지우고 다시 설치하는 과정에서 mysqlclient 또는 가상환경의 Django와의 연결에 문제가 생겼다고 짐작했습니다. 따라서 NameError를 해결하기 위해 현재 MySQL이 깔려있는 상태에서 가상환경을 새로 만들고 mysqlclient를 다시 설치했습니다. 새로운 가상환경을 통해 python manage.py migrate와 python manage.py runserver을 시도해보니 드디어 서버가 켜지더군요.

안타깝게도 저도 "ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)”가 왜 사라졌는지에 대해서 완전히 이해하지 못하고 있습니다. 제가 어제 시도했던 키워드들은 이 스택오버플로우 (https://stackoverflow.com/q/15450091)의 답변들에서 찾았고, 그 후에는 brew와 공식 파일을 통해 MySQL을 완전히 지우고 새로 까는 과정을 거쳤습니다. 큰 도움이 되지 못한 것 같아 죄송하네요 ㅠㅠ
